src/game/object_helpers.c

```diff
@@ -130,16 +130,21 @@ Gfx *geo_switch_area(s32 callContext, struct GraphNode *node, UNUSED void *conte
         if (gMarioObject == NULL) {
             switchCase->selectedCase = 0;
         } else {
-#ifdef ENABLE_VANILLA_LEVEL_SPECIFIC_CHECKS
-            if (gCurrLevelNum == LEVEL_BBH) {
-                // In BBH, check for a floor manually, since there is an intangible floor. In custom hacks this can be removed.
+#ifdef VERTICAL_ROOMS
```

Comment:
is there any issue with checking if `gCurrentArea->surfaceRooms != NULL` instead of using this define? i feel like there has to be a way if checking that an area has rooms before doing this check, that way we dont need to have a define for this and its automatic

Reply:
good point, I'll see if that works and if so I'll remove the define

Reply:
another consideration we're having: add a flag whether or not an area has `surface intangible` floors, then if `flag` then check